lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/HttpTrackingURLProvider.java

```diff
@@ -62,11 +62,36 @@ public String getTrackingUrl() {
       return "";
     }
 
+    return getHost() + ":" + getPort();
+  }
+
+  /**
+   * get host of the driver.
+   *
+   * @return
+   */
+  @Override
+  public String getHost() {
+
     try {
-      return InetAddress.getLocalHost().getHostAddress() + ":" + httpServer.getPort();
+      return InetAddress.getLocalHost().getHostAddress();
     } catch (final UnknownHostException e) {
       LOG.log(Level.WARNING, "Cannot get host address.", e);
       throw new RuntimeException("Cannot get host address.", e);
     }
   }
+
+  /**
+   * get port number of the http server.
+   *
+   * @return
```

Comment:
Even LocalAddressProvider has API to return local host, but it essentially is a configuration provider.
I will refactor the code to make HttpTrackingURLProvider nicer.